packages/api/src/task/scheduler.ts

```diff
@@ -152,6 +152,7 @@ export class TaskScheduler {
               ipfs: {
                 spec: inputAsset.storage.ipfs.spec,
                 ...taskOutputToIpfsStorage(event.output.export.ipfs),
+                createdAt: task.createdAt,
```

Comment:
The `createdAt` will get the time when the `task` started, but not when it actually finished saving the file to IPFS. So we should use the task `updatedAt` here which is updated in the end of this function on line 169. We can create an `updatedAt` variable before this whole `switch` case adn then use it both here and in the update on 169.

Reply:
Also, this field should be created inside `taskOutputToIpfsStorage`. It creates the `nftMetadata` field which should also have the `updatedAt` field set to the right value (not necessarily the same of the "root" IPFS export).

Another approach is not to have the field in the `nftMetadata`, and I'd be OK with that as well. But in this case you should add the field not to the `ipfs-file-info` schema, but to the `asset.storage.ipfs` object instead. (Right below that line I linked you with the `$ref`).

Reply:
Agree on not having it inside the `nftMetadata` which seems more about the metadata per-se. So I will fix the schema.

I just disagree on defining `updatedAt = Date.now()` before the switch, given it's going to be used in different places that require the finishing of async functions. I would just fix my line with `updatedAt: Date.now()`. The same way it is currently used at 169. So both values are actually correct.

Reply:
That works as well! We just have tried using a consistent timestamp on DB updates that are conceptually a transaction (but aren't right now). All of the async operations here are DB updates (and maybe sending events, which conceptually also shouldn't change the timestamps)

When we do create transactions like that, all of these updates will effectively happen at the same time on the DB, so we used consistent timestamps since the beginning here. A side benefit is that you can also correlate the asset updated time with the task that finished at that time.

I'm fine with using the timestamp that we actually sent that specific update query tho. Just a different interpretation of the field which also makes sense.

Reply:
Thanks for the explanation